extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/redirect/RedirectTest.java

```diff
@@ -31,50 +31,66 @@ public class RedirectTest {
 
     @Test
     void shouldRedirect3Times_whenMax4() {
-        RedirectingResourceClient client = RestClientBuilder.newBuilder()
+        RedirectingResourceClient302 client = RestClientBuilder.newBuilder()
                 .baseUri(uri)
                 .followRedirects(true)
                 .property(QuarkusRestClientProperties.MAX_REDIRECTS, 4)
-                .build(RedirectingResourceClient.class);
+                .build(RedirectingResourceClient302.class);
         Response call = client.call(3);
         assertThat(call.getStatus()).isEqualTo(200);
     }
 
     @Test
     void shouldNotRedirect3Times_whenMax2() {
-        RedirectingResourceClient client = RestClientBuilder.newBuilder()
+        RedirectingResourceClient302 client = RestClientBuilder.newBuilder()
                 .baseUri(uri)
                 .followRedirects(true)
                 .property(QuarkusRestClientProperties.MAX_REDIRECTS, 2)
-                .build(RedirectingResourceClient.class);
-        assertThat(client.call(3).getStatus()).isEqualTo(307);
+                .build(RedirectingResourceClient302.class);
+        assertThat(client.call(3).getStatus()).isEqualTo(302);
     }
 
     @Test
     void shouldNotRedirectOnPostMethodsByDefault() {
-        RedirectingResourceClient client = RestClientBuilder.newBuilder()
+        RedirectingResourceClient302 client302 = RestClientBuilder.newBuilder()
                 .baseUri(uri)
                 // this property should be ignored in POST
                 .followRedirects(true)
-                .build(RedirectingResourceClient.class);
-        assertThat(client.post().getStatus()).isEqualTo(307);
+                .build(RedirectingResourceClient302.class);
+        assertThat(client302.post().getStatus()).isEqualTo(302);
+
+        RedirectingResourceClient307 client307 = RestClientBuilder.newBuilder()
+                .baseUri(uri)
+                // this property should be ignored in POST
+                .followRedirects(true)
```

Comment:
To be honest, I am not completely sure about this when the response is 307...

Reply:
You mean that we should enforce it when it's 307?

Reply:
Thats what I am wondering... I don't know that the proper answer is...